src/planner/binder/expression/bind_lambda.cpp

```diff
@@ -35,74 +35,63 @@ idx_t GetLambdaParamIndex(const vector<DummyBinding> &lambda_bindings, const Bou
 	return offset;
 }
 
-BindResult ExpressionBinder::BindExpression(LambdaExpression &expr, idx_t depth, const LogicalType &list_child_type,
-                                            optional_ptr<bind_lambda_function_t> bind_lambda_function) {
+void ExtractParameterList(LambdaExpression &expr) {
 
-	// this is for binding JSON
-	if (!bind_lambda_function) {
-		auto lhs_expr = expr.lhs->Copy();
-		OperatorExpression arrow_expr(ExpressionType::ARROW, std::move(lhs_expr), expr.expr->Copy());
-		return BindExpression(arrow_expr, depth);
-	}
+	// extract the lambda parameter list, which is a single column
+	// reference, or a list of column references
 
-	// binding the lambda expression
 	D_ASSERT(expr.lhs);
-	if (expr.lhs->expression_class != ExpressionClass::FUNCTION &&
-	    expr.lhs->expression_class != ExpressionClass::COLUMN_REF) {
-		throw BinderException(
-		    "Invalid parameter list! Parameters must be comma-separated column names, e.g. x or (x, y).");
-	}
+	bool valid_param_list = true;
 
-	// move the lambda parameters to the params vector
 	if (expr.lhs->expression_class == ExpressionClass::COLUMN_REF) {
+		// single column reference
 		expr.params.push_back(std::move(expr.lhs));
-	} else {
+
+	} else if (expr.lhs->expression_class == ExpressionClass::FUNCTION) {
+		// list of column references
 		auto &func_expr = expr.lhs->Cast<FunctionExpression>();
-		for (idx_t i = 0; i < func_expr.children.size(); i++) {
-			expr.params.push_back(std::move(func_expr.children[i]));
+		for (auto &column_ref_expr : func_expr.children) {
+			if (column_ref_expr->expression_class != ExpressionClass::COLUMN_REF) {
+				valid_param_list = false;
+				break;
+			}
+			expr.params.push_back(std::move(column_ref_expr));
```

Comment:
Should we check if the column names are qualified or not or does this happen elsewhere? Currently this throws an exception, does it still do that?

```sql
select list_transform([1,2,3], sqrt(xxx.z) -> xxx.z + 1) as l ;
```